apps/auth0-rules/src/on-execute-post-login.ts

```diff
@@ -70,53 +64,56 @@ const extractUser = (response: Auth0UserResponse): User | gp2Auth.User => {
   };
 };
 
-const addUserMetadata: Rule<{ invitationCode: string }> = async (
-  auth0User,
-  context,
-  callback,
+export const onExecutePostLogin = async (
+  event: Auth0PostLoginEventWithSecrets,
+  api: Auth0PostLoginApi,
 ) => {
-  const redirect_uri = new URLSearchParams(context.request.query).get(
+  const redirect_uri = new URLSearchParams(event.request.query).get(
     'redirect_uri',
   )
-    ? new URLSearchParams(context.request.query).get('redirect_uri')
-    : context.request.body.redirect_uri;
+    ? new URLSearchParams(event.request.query).get('redirect_uri')
+    : event.request.body.redirect_uri;
   if (!redirect_uri) {
-    return callback(new Error('Missing redirect_uri'));
+    return api.access.deny('Missing redirect_uri');
   }
 
   const prUrlRegex = new RegExp(
-    `https://(?<pr_number>[0-9]+).${configuration.APP_DOMAIN}`,
+    `https://(?<pr_number>[0-9]+).${event.secrets.PR_APP_DOMAIN}`,
   );
   const matches = prUrlRegex.exec(redirect_uri);
-
   const apiURL = matches?.groups?.pr_number
-    ? `https://api-${matches.groups.pr_number}.${configuration.APP_DOMAIN}`
-    : configuration?.APP_ORIGIN;
-  const apiSharedSecret = configuration?.API_SHARED_SECRET;
-
+    ? `https://api-${matches.groups.pr_number}.${event.secrets.PR_APP_DOMAIN}`
+    : event.secrets.ASAP_API_URL;
   try {
-    const response = await got(`${apiURL}/webhook/users/${auth0User.user_id}`, {
-      headers: {
-        Authorization: `Basic ${apiSharedSecret}`,
+    const response = await got(
+      `${apiURL}/webhook/users/${event.user.user_id}`,
+      {
+        headers: {
+          Authorization: `Basic ${event.secrets.AUTH0_SHARED_SECRET}`,
+        },
+        timeout: 10000,
       },
-      timeout: 10000,
-    }).json<Auth0UserResponse>();
-
+    ).json<Auth0UserResponse>();
     if (isUserMetadataResponse(response) && response.alumniSinceDate) {
-      return callback(new UnauthorizedError('alumni-user-access-denied'));
+      return api.access.deny('alumni-user-access-denied');
     }
-
     const user = extractUser(response);
-
-    context.idToken[new URL('/user', redirect_uri).toString()] = user;
-    // Uncomment for dev auth0. This allows pointing to dev api from local FE
-    // context.idToken[new URL('/user', 'https://dev.hub.asap.science').toString()] = user;
-
-    return callback(null, auth0User, context);
+    api.idToken.setCustomClaim(new URL('/user', redirect_uri).toString(), user);
+    if (event.secrets.AUTH0_ADDITIONAL_CLAIM_DOMAIN) {
+      api.idToken.setCustomClaim(
+        new URL(
+          '/user',
+          event.secrets.AUTH0_ADDITIONAL_CLAIM_DOMAIN,
+        ).toString(),
+        user,
+      );
+    }
   } catch (err) {
-    const error = handleError(err);
-    return callback(error);
+    let errorMessage;
+    if (err instanceof Error) {
+      errorMessage = err.message;
+    }
+    return api.access.deny(errorMessage ?? 'Unexpected Error');
```

Comment:
```suggestion
    let errorMessage;
    if (err instanceof Error) {
      errorMessage = err.message;
    }
    return api.access.deny(err instanceof Error ? errorMessage.err : 'Unexpected Error');
```
to me this still reads ok and avoids let

Reply:
Its test coverage I was trying to avoid. Writing it this way would require me to write mocks for got instead of using Nock for a test case demanded by typescript  I doubt will actually happen.